### DISCUSSION TITLE
Add the ability to choose the WF version when starting WF by MQ

### DISCUSSION THREAD

**Discussion by omriv** [Netflix/conductor#3650] [Ideas]
Currently, when starting a WF by HTTP you can provide the requested WF version, but when starting a WF using an Event Handler, you have to create a dedicated Event Handler which contains a hardcoded version of the WF (0 for latest).
When working with several WF versions (mostly in dev environments) we have to create dedicated Event Handlers for each WF version, and it also forces the caller system to know all the routing keys for all the versions (We are using RabbitMQ).

 My idea is to consolidate the two ways of starting a WF, by adding the ability to pass the WF version on the message so the Event Handler would be able to take it from there.

What do you think?

**Comment by v1r3n** [comment]
I think that is a great enhancement.

**Comment by YevgenyMe** [comment]
Sounds like a Great Upgrade ! Could be Very Useful ! ! !
